src/Model/TrackedManyManyList.php

```diff
@@ -34,27 +36,50 @@ protected function recordManyManyChange($type, $item)
         }
         $parts = explode('_', $joinName);
         if (isset($parts[0]) && count($parts) > 1) {
-
             // table name could be sometihng like Symbiote_DataChange_Tests_TestObject_Kids
             // which is ClassName_RelName, with
-            array_pop($parts);
-
-            $addingToClass = implode('\\', $parts);
-            $addingTo = $this->getForeignID();
-            
-            if (class_exists($addingToClass)) {
-                $onItem = $addingToClass::get()->byID($addingTo);
-                if ($onItem) {
-                    if ($item && !($item instanceof DataObject)) {
-                        $class = $this->dataClass;
-                        $item = $class::get()->byID($item);
-                    }
-                    $join = $type == 'add' ? ' to ' : ' from ';
-                    $type = ucfirst($type) . ' "' . $item->Title . '"' . $join . $parts[1];
-                    $onItem->RelatedItem = $item->ClassName . ' #' . $item->ID;
-                    singleton('DataChangeTrackService')->track($onItem, $type);
-                }
+            $tableName = $parts;
+            $relationName = array_pop($tableName);
+            $tableName = implode('_', $tableName);
+
+            $addingToClass = $this->tableClass($tableName);
+            if (!$addingToClass) {
+                return;
+            }
+            if (!class_exists($addingToClass)) {
+                return;
             }
+            $onItem = $addingToClass::get()->byID($this->getForeignID());
+            if (!$onItem) {
+                return;
+            }
+            if ($item && !($item instanceof DataObject)) {
+                $class = $this->dataClass();
+                $item = $class::get()->byID($item);
+            }
+            $join = $type === 'add' ? ' to ' : ' from ';
+            $type = ucfirst($type) . ' "' . $item->Title . '"' . $join . $relationName;
+            $onItem->RelatedItem = $item->ClassName . ' #' . $item->ID;
+            singleton('DataChangeTrackService')->track($onItem, $type);
+        }
+    }
+
+    /**
+     * Find the class for the given table.
+     *
+     * Stripped down version from framework that does not attempt to strip _Live and _versions postfixes as
+     * that throws errors in its preg_match(). (At least it did as of 2018-06-22 on SilverStripe 4.1.1)
+     *
+     * @param string $table
+     * @return string|null The FQN of the class, or null if not found
+     */
+    private function tableClass($table)
+    {
+        $tables = DataObject::getSchema()->getTableNames();
```

Comment:
How does this work when something isn't using a `table_name` for some reason? It's a little hard to follow the code, but I guess I'm just trying to figure out how this works when a namespace is/isn't present compared to how it worked previously.

Reply:
I'm assuming this is where the with/without underscores testing comes into play. In which case, I don't see any issues here. Happy for this to be merged and tagged.

Reply:
The issue with the old code was it tried to figure out the class name from the table name (That's how the SS 3.X code does it). This straight up wouldn't work in some cases. So I fixed the lookup to use core SS functionality.